src/main/flight/autopilot.c

```diff
@@ -70,9 +137,9 @@ void altitudeControl(float targetAltitudeCm, float taskIntervalS, float vertical
     const float hoverOffset = autopilotConfig()->hover_throttle - PWM_RANGE_MIN;
 
     float throttleOffset = altitudeP + altitudeI - altitudeD + altitudeF + hoverOffset;
-    const float tiltMultiplier = 2.0f - fmaxf(getCosTiltAngle(), 0.5f);
-    // 1 = flat, 1.24 at 40 degrees, max 1.5 around 60 degrees, the default limit of Angle Mode
-    // 2 - cos(x) is between 1/cos(x) and 1/sqrt(cos(x)) in this range
+    const float tiltMultiplier = 1.0f / fmaxf(getCosTiltAngle(), 0.5f);
```

Comment:
This assumes linear throttle-thrust relationship. It will probably overcompensate.

Reply:
I have tested it, and it works a lot better.
The other method typically lost altitude every time the quad did a hard stop to 40-50 degrees.  With this change it is much more stable.
Props tend to develop stronger thrust at lower RPM, so if anything we would need to provide more of a throttle increase when pushing throttle up, than less.
In any case we have per-motor thrust linearisation code for people who wish to experiment with linearising thrust.
And, as a side-effect of this change, which applies more throttle when we tilt hard to stop the quad, the XY thrust goes up as the throttle goes up, making the stop cleaner and quicker :-)  It's much better.

Reply:
@ledvinap  - update, with recent testing...

In altitude hold mode, i.e. in Angle mode with altitude control from GPS, pitching forward or back with this change is associated with incredibly stable altitude.  It hardly shifts off the altitude target despite aggressive inputs.

In position hold, however, a hard pull-back from the position hold algorithm results in transient lift.  I don't really know why it happens more in this situation than while in manual pitch angle control.   It could be the lack of an upsampling filter in the PIDs to remove the 10Hz spikes, since a hard position hold pull-back is associated with a lot of 10Hz noise in the motors.  I'll test further once the antialiasing filter is in place.

Reply:
This is an image from a log where I am in altitude hold mode and make some quick pitch requests (purple dips in top trace).  
Motor responses are from the tilt correction.
Altitude was extremely stable with transient pitch angles of 40 degrees approx.
<img width="1110" alt="Screen Shot 2024-10-28 at 8 51 41 am" src="https://github.com/user-attachments/assets/921014e7-f8a1-4227-bc9e-a2040e61a4cd">

Here is some auto-stops from position hold, with quite aggressive D in position hold.
To the left of the red centre line I accelerate the quad in Angle mode by requesting pitch forward.
There is no altitude change while doing this.
Then at centre I release, and position hold then brakes hard, and corrects attitude.
There is a big spike in motors from the spike in D and then subsequently the 10Hz stuff probably adds to average motor output.
This is something I have to look into in more detail.  But, for sure, the full cosine tilt correction is perfect, in itself.

<img width="1110" alt="Screen Shot 2024-10-28 at 8 52 56 am" src="https://github.com/user-attachments/assets/82a627d4-bd85-430d-88b7-955c91e2a28d">